h5py/_objects.templ.pyx

```diff
@@ -221,10 +221,17 @@ cdef class ObjectID:
             registry[self._pyid] = weakref.ref(self)
 
 
-    def __dealloc__(self):
+    def __del__(self):
+        self._del()
+
+    # Hack to increase the reference counter of _phil by 1.
+    # In free-threading interpreters, this prevents a race condition during interpreter
+    # shutdown where the global _phil is dereferenced by the main thread while
+    # ObjectID.__del__ is being executed in another thread.
```

Comment:
Is it still problematic to call `__dealloc__` if you have a strong reference? I thought you only had to either have a strong reference or switch from `__dealloc__` to `__del__` to avoid the race at shutdown. In this PR you're doing both of those.

Reply:
+1